Documentation/design-docs/AssemblyLoadContext.Activate.md

```diff
@@ -0,0 +1,306 @@
+# AssemblyLoadContext.Activate
+
+The `AssemblLoadContext` is designed to create a assembly load scope and type resolving scope.
+
+It is intended as a replacement for the isolation offered by `AppDomain`s.
+
+Current APIs affect and/or use a given `AssemblyLoadContext` by:
+* Operating directly on the `AssemblyLoadContext`
+* Operating on an assembly loaded into the `AssemblyLoadContext`
+* Executing methods defined within an assembly loaded by the `AssemblyLoadContext`
+
+There is a need for a general solution to allow code executing outside a given `AssemblyLoadConxt` to behave as if it was loaded within the context.
+
+This is in part because of the design of `AssemblyLoadContext` which
+* Requires `System.Private.CoreLib` to be loaded exclusively in the `Default` context.
+* Strongly recommends the runtime framework be completely loaded into the `Default` context.
+* Recommends most frameworks be loaded exclusively in the `Default` context.
+
+This proposal is for a mechanism for code to explicitly set a specific `AssemblyLoadContext` as the active context.
+
+## Proposed API extension(s)
+```C#
+namespace System.Runtime
+{
+  partial class AssemblyLoadContext
+  {
+    // When set supersedes implicit context as determine by callers assembly.
+    // Affects most/all APIs which are (or will be) marked [System.Security.DynamicSecurityMethod]
+    // Implemented as AsyncLocal<AssemblyLoadContext> to allow setting to follow async & threads
+    public static AssemblyLoadContext ActiveContext { get { return _asyncLocalActiveContext.Value; }}
+
+    // Allow client to explicitly set the active context for a using block
+    public IDisposable Activate();
```

Comment:
This will always allocate. It won't be friendly for chatty APIs that need to call this a lot. (I do not know whether it is a problem - just pointing it out.)

Reply:
A struct-based disposable is another option if we decide we need to reduce allocations.

Reply:
Added a note to both effects in the doc